test/elementhandle.spec.js

```diff
@@ -248,4 +248,25 @@ describe('ElementHandle specs', function() {
       }
     });
   });
+
+  describe('Custom queries', function() {
+    it('should register and unregister', async() => {
+      const {page, puppeteer} = getTestState();
+      await page.setContent('<div id="not-foo"></div><div id="foo"></div>');
+
+      // Register.
+      puppeteer.registerCustomQueryFunction('get-by-id', (element, selector) => document.querySelector(`[id="${selector}"]`));
```

Comment:
I think it'd be nice to limit query handler names to `/^[a-zA-Z]+$/`, matching built-in JavaScript APIs. We could validate this upon registration. WDYT? In this case, it'd be `getById` instead of `get-by-id`.

Reply:
I think  using `id:elementId` will be better than `get-by-id/elementId`. 

<img width="475" alt="图片" src="https://user-images.githubusercontent.com/16131917/80323134-193cb880-885c-11ea-9060-cbb85b6cd541.png">

Reply:
When registering your own query handler you could name it `id` and then it would be `id/elementId`, which seems close enough?

The reason we cannot use `:` is because it has special meaning in CSS, so selectors like `p:first` become ambiguous.